### PR TITLE
Add heart beat age metric.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/feedhandler.h [hunk 1]
```diff
@@ -96,6 +96,8 @@ class FeedHandler: private search::transactionlog::client::Callback,
     std::mutex                             _syncLock;
     SerialNum                              _syncedSerialNum; 
     bool                                   _allowSync; // Sanity check
+    mutable std::mutex                     _heart_beat_time_lock;
+    vespalib::steady_time                  _heart_beat_time;
```
Comment (baldersheim) [inline searchcore/src/vespa/searchcore/proton/server/feedhandler.h:100]: Can't you make this an atomic to avoid the separate extra lock ?

**Review by baldersheim — CHANGES_REQUESTED** [review]
Consider using an atomic instead

File: searchcore/src/vespa/searchcore/proton/metrics/documentdb_tagged_metrics.cpp [hunk 2]
```diff
@@ -272,6 +272,7 @@ DocumentDBTaggedMetrics::DocumentDBTaggedMetrics(const vespalib::string &docType
       bucketMove(this),
       totalMemoryUsage(this),
       totalDiskUsage("disk_usage", {}, "The total disk usage (in bytes) for this document db", this),
+      heart_beat_age("heart_beat_age", {}, "How long ago heart beat maintenace job was run", this),
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/metrics/documentdb_tagged_metrics.cpp:275]: Please add info on unit used: "ago (in seconds) heart"

**Review by baldersheim — APPROVED** [review]
So steady_time can not be atomic then ?

File: searchcore/src/vespa/searchcore/proton/server/feedhandler.h [hunk 3]
```diff
@@ -96,6 +96,7 @@ class FeedHandler: private search::transactionlog::client::Callback,
     std::mutex                             _syncLock;
     SerialNum                              _syncedSerialNum; 
     bool                                   _allowSync; // Sanity check
+    std::atomic<uint64_t>                  _heart_beat_time;
```
Comment (baldersheim) [inline searchcore/src/vespa/searchcore/proton/server/feedhandler.h:100]: I do see that we have test for atomic time.
static_assert(std::atomic<steady_time>::is_always_lock_free, "std::atomic<steady_time> should be atomic");
Then I suggest that you use std::atomic<steady_time>. Then you avoid all the casting.

**Review by baldersheim — APPROVED** [review]
Perfect

**Pull request merged**